src/generic/eigen_solver.cc

```diff
@@ -700,19 +736,205 @@ namespace oomph
+    unsigned eval_count = 0;
+    while (eval_count < n)
+    {
+      // i-th eigenvalue is real:
+      if (alpha_eval[eval_count].imag() == 0.0)
+      {
+        // Resize the single eigenvector associated with this
+        // single real eigenvalue
+        eigenvector[eval_count].resize(n);
+        for (unsigned j = 0; j < n; ++j)
+        {
+          eigenvector[eval_count][j] =
+            std::complex<double>(eigenvector_aux[j][eval_count], 0.0);
+        }
+        eval_count++;
+      }
+      // Assume (and check!) that complex conjugate pairs follow each other
+      // as implied by
+      // http://www.netlib.org/lapack/explore-html/d9/d8e/group__double_g_eeigen_ga4f59e87e670a755b41cbdd7e97f36bea.html
+      else
+      {
+#ifdef PARANOID
+
+        // Are the eigenvalues finite?
+        if ((beta_eval[eval_count] != 0.0) &&
+            (beta_eval[eval_count + 1] != 0.0))
+        {
+          std::complex<double> lambda_this =
+            alpha_eval[eval_count] / beta_eval[eval_count];
+
+          std::complex<double> lambda_next =
+            alpha_eval[eval_count + 1] / beta_eval[eval_count + 1];
+
+          // Check cc-ness to within tolerance
+          if (fabs(lambda_this.imag() + lambda_next.imag()) >
+              Tolerance_for_ccness_check)
+          {
+            std::ostringstream error_stream;
+            error_stream << "Non-zero imaginary part of eigenvalue "
+                         << eval_count << " : " << lambda_this.imag()
+                         << std::endl;
+            error_stream
+              << "isn't the negative of its subsequent value       : "
+              << lambda_next.imag() << std::endl
+              << "Their sum " << (lambda_this.imag() + lambda_next.imag())
+              << " is greater than Tolerance_for_ccness_check = "
+              << Tolerance_for_ccness_check << std::endl;
+            throw OomphLibError(error_stream.str(),
+                                OOMPH_CURRENT_FUNCTION,
+                                OOMPH_EXCEPTION_LOCATION);
+          }
+          if (fabs(lambda_this.real() - lambda_next.real()) >
+              Tolerance_for_ccness_check)
+          {
+            std::ostringstream error_stream;
+            error_stream << "Real parts of complex eigenvalue  " << eval_count
+                         << " : " << lambda_this.real() << std::endl;
+            error_stream
+              << " doesn't agree with its supposed-to-be cc counterpart     : "
+              << lambda_next.real() << std::endl
+              << "Their difference "
+              << (lambda_this.real() - lambda_next.real())
+              << " is greater than Tolerance_for_ccness_check = "
+              << Tolerance_for_ccness_check << std::endl;
+            throw OomphLibError(error_stream.str(),
+                                OOMPH_CURRENT_FUNCTION,
+                                OOMPH_EXCEPTION_LOCATION);
+          }
+        }
+        else
```

Comment:
Should this else block be removed?

Reply:
The commented out oomph-info still labels the logic, so I kept it for that reason, but fair enough. Changed the comment at the very top and removed the else block.

Done.